### PR TITLE
5.9 kernel driver changes: make_request_fn removed, NR_SLAB_RECLAIMABLE changed, others...

### PR DESCRIPTION
**Pull request by ckane**
The latest merges to the linux staging-testing branch introduced some incompatibilities due to API changes. In particular, `NR_SLAB_RECLAIMABLE` was renamed to `NR_SLAB_RECLAIMABLE_B`, the whole `make_request_fn` interface was replaced with a new `submit_bio` member added to the `struct block_device_operations` structure, and a number of block-device specific function prototypes were moved from `linux/fs.h` into `linux/blkdev.h`.

### Motivation and Context

These changes fix the breakage caused to the build from 3 recent API changes merged into the Linux 5.9 development tree: `NR_SLAB_RECLAIMABLE_B` introduction, removal of `make_request_rn`, and moving of a number of exported API prototypes into `linux/blkdev.h`.

### Description

The `autoconf` macros under `config/` had to be updated to search for and take into account these 3 breaking API changes that were introduced in Linux 5.9. Additionally, macro tests within the code were also added to conditionally compile the changes in the appropriate kernel targets.

### How Has This Been Tested?

I have compiled this against the following kernels (on Arch Linux):

- 5.7.12 (standard linux kernel package)
- 5.8.0 (AUR, 5.8.0-1-mainline)
- 5.9 (linux staging-testing branch current as of 2020-08-09)
- 5.9 (torvalds' `master` branch as of 2020-08-09 also compiles properly)

I have not run it on a live system/VM yet. I plan to install Arch on a VM with these kernels to do a run-time test. Right now I've been using ZFS 0.8.4 on my personal system, so don't have a reference to compare for using the latest code on a ZFS system yet. Since I got it to compile I wanted to share the PR to get the benefit of a broader set of tests from users, while I set up my test environment.

### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [X] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [X] I have updated the documentation accordingly.
- [X] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [X] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by ckane** [comment]
I also now have this running in a VM on Arch 2020.8 with a 5.7.12 kernel.

File: config/kernel-make-request-fn.m4 [hunk 1]
```diff
@@ -26,57 +26,75 @@ AC_DEFUN([ZFS_AC_KERNEL_SRC_MAKE_REQUEST_FN], [
 		struct request_queue *q __attribute__ ((unused));
 		q = blk_alloc_queue(make_request, NUMA_NO_NODE);
 	])
+
+	ZFS_LINUX_TEST_SRC([block_device_operations_submit_bio], [
+		#include <linux/blkdev.h>
+	],[
+        struct block_device_operations o;
+        o.submit_bio = NULL;
```
Comment (behlendorf) [inline config/kernel-make-request-fn.m4:34]: nit: indentation

**Comment by ckane** [comment]
How does the backporting work in this project? Would you want me to make another PR against the 0.8 branch or do will that just be handled during an upcoming backporting wave?

**Comment by behlendorf** [comment]
@ckane normally we'll cherry pick the commit from master and adapt it as needed when putting together the next point release.  No need to open a new PR.

**Pull request closed without merging**